compiler/src/dotty/tools/dotc/util/HashSet.scala

```diff
@@ -121,6 +121,13 @@ class HashSet[T](initialCapacity: Int = 8, capacityMultiple: Int = 2) extends Mu
       idx = nextIndex(idx)
       e = entryAt(idx)
 
+  def remove(x: T): Boolean =
+    if contains(x) then
+      this -= x
+      true
+    else
+      false
```

Comment:
@smarter This is fairly inefficient. Should I move the implem of `-=` inside `remove` and implement `-=` in terms of `remove` instead?

Reply:
Sounds good to me.